web-dev/src/components/UserProfile/index.js

```diff
@@ -0,0 +1,40 @@
+import React from "react"
+import UserProfileComponent from "./UserProfileComponent"
+import NavigationTopBar from "../NavigationTopBar/index.js";
+import ProfileComments from "../ProfileComments"
+import profile from "../data/profile.json"
+
+import {Link, useLocation} from "react-router-dom";
+import "../Profile/profile.css"
+
+const UserProfile = () => {
+    const otherUser = useLocation().pathname.split(/[/:]+/)[2];
```

Comment:
I guess the work that left to be done here is to go to the database and extract the user's data that has username equals to the const otherUser here. Then we can use that profile to render the UserProfileComponent file. I'm still not sure how to do this with simply useEffect instead of doing all the reducers and actions set up like in HW8. I would need help here.